src/node_options.cc

```diff
@@ -408,6 +416,11 @@ EnvironmentOptionsParser::EnvironmentOptionsParser() {
             &EnvironmentOptions::force_async_hooks_checks,
             kAllowedInEnvironment,
             true);
+  AddOption(
```

Comment:
I'm sorry I did not pick up on this earlier. At some point last year we added easier handling for boolean options. This is an example of another one

```JavaScript
  AddOption("--force-async-hooks-checks",
            "disable checks for async_hooks",
            &EnvironmentOptions::force_async_hooks_checks,
            kAllowedInEnvironment,
            true);
```

That adds 2 possible flags

```shell
--force-async-hooks-checks
--no-force-async-hooks-checks
```

I think we should use the same technique to add the boolean option we are adding. In that context what likely makes sense is

```JavaScript
  AddOption("--node-api-uncaught-exceptions-policy",
            "disable checks for async_hooks",
            &EnvironmentOptions::force_async_hooks_checks,
            kAllowedInEnvironment,
            true);
```
This would make the option on by default and provide the following flags

```
--node-api-uncaught-exceptions-policy
--no-node-api-uncaught-exceptions-policy
```

We can switch the default by just changing the true/false in the part that adds the option in we are forced to fall back to older behavior.